OptimoveSDK/optimove-sdk/src/main/java/com/optimove/sdk/optimove_sdk/kumulos/OptimobileConfig.java

```diff
@@ -0,0 +1,275 @@
+import androidx.annotation.DrawableRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.optimove.sdk.optimove_sdk.R;
+
+import org.json.JSONObject;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Represents the configuration for the Kumulos client
+ */
+public final class OptimobileConfig {
+
+    @DrawableRes
+    static final int DEFAULT_NOTIFICATION_ICON_ID = R.drawable.kumulos_ic_stat_notifications;
+    static final int DEFAULT_SESSION_IDLE_TIMEOUT_SECONDS = 23;
+
+    private String apiKey;
+    private String secretKey;
+
+    private String optimoveToken;
+    private String optimoveConfigFile;
+
+    @DrawableRes
+    private int notificationSmallIconId;
+    private InAppConsentStrategy inAppConsentStrategy;
+    private int sessionIdleTimeoutSeconds;
+
+    private JSONObject runtimeInfo;
+    private JSONObject sdkInfo;
+    private Map<UrlBuilder.Service, String> baseUrlMap;
+
+    private URL deepLinkCname;
+    private DeferredDeepLinkHandlerInterface deferredDeepLinkHandler;
+
+    public enum InAppConsentStrategy{
+        AUTO_ENROLL,
+        EXPLICIT_BY_USER
+    }
+
+    // Private constructor to discourage not using the Builder.
+    private OptimobileConfig() {}
+
+    private void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    private void setSecretKey(String secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    private void setOptimoveToken(String optimoveToken) {
+        this.optimoveToken = optimoveToken;
+    }
+
+    private void setOptimoveConfigFile(String optimoveConfigFile) {
+        this.optimoveConfigFile = optimoveConfigFile;
+    }
+
+    private void setNotificationSmallIconId(@DrawableRes int notificationSmallIconId) {
+        this.notificationSmallIconId = notificationSmallIconId;
+    }
+
+    private void setSessionIdleTimeoutSeconds(int timeoutSeconds) {
+        this.sessionIdleTimeoutSeconds = timeoutSeconds;
+    }
+
+    private void setRuntimeInfo(JSONObject info) {
+        this.runtimeInfo = info;
+    }
+
+    private void setSdkInfo(JSONObject info) {
+        this.sdkInfo = info;
+    }
+
+    private void setBaseUrlMap(Map<UrlBuilder.Service, String> baseUrlMap) {
+        this.baseUrlMap = baseUrlMap;
+    }
+
+    private void setInAppConsentStrategy(InAppConsentStrategy strategy) {
+        this.inAppConsentStrategy = strategy;
+    }
+
+    private void setCname(@Nullable URL deepLinkCname) {
+        this.deepLinkCname = deepLinkCname;
+    }
+    private void setDeferredDeepLinkHandler(DeferredDeepLinkHandlerInterface deferredHandler) {
+        this.deferredDeepLinkHandler = deferredHandler;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public String getSecretKey() {
+        return secretKey;
+    }
+
+    public String getOptimoveToken() {
+        return optimoveToken;
+    }
+
+    public String getOptimoveConfigFile() {
```

Comment:
shall we rename some of the methods here?
   
   - getOptimoveConfigFile ---> getOptimoveConfigFileName
   - getApiKey ---> getKumulosApiKey
   - getSecretKey ---> getKumulosSecretKey

shall we also make them package-private?

Reply:
we definitely dont want to start adding references to Kumulos, but we can likely make these package private so long as all the necessary consumers are given them during init.

Reply:
`getKumulosApiKey` etc come from the API you sent on slack, i thought the intention was to have it clear as whose credentials are where. ok, will have as `getConfigFileName, getApiKey and getSecretKey`